Makefile

```diff
@@ -81,16 +81,23 @@ DEPS += $(OBJS:.o=.d)
 MKDIR_P ?= @mkdir -p
 TARGET ?= linux
 $(info TARGET [$(TARGET)])
+OPENSSL_LIB_DIR = $(TOP_DIR)/framework/openssl/openssl-OpenSSL_1_1_1w/build/lib/
 
 # defaults for target arm
 ifeq ($(TARGET),arm)
 #CC := arm-rdk-linux-gnueabi-gcc -mthumb -mfpu=vfp -mcpu=cortex-a9 -mfloat-abi=soft -mabi=aapcs-linux -mno-thumb-interwork -ffixed-r8 -fomit-frame-pointer
 # CFLAGS will be overriden by Caller as required
 INC_DIRS += $(UT_DIR)/sysroot/usr/include
-OPENSSL_LIB_DIR = $(TOP_DIR)/framework/openssl/openssl-OpenSSL_1_1_1w/build/lib/
 XLDFLAGS += $(OPENSSL_LIB_DIR)/libssl.a $(OPENSSL_LIB_DIR)/libcrypto.a -ldl
 else
-XLDFLAGS += -lssl -lcrypto -lz
+#linux case
+# Check if the directory exists
+ifneq ($(wildcard $(OPENSSL_LIB_DIR)),)
+XLDFLAGS += $(OPENSSL_LIB_DIR)/libssl.a $(OPENSSL_LIB_DIR)/libcrypto.a -ldl
```

Comment:
`-ldl` we're not building this, so don't we therefore need too if someone requires it...

Reply:
The -ldl flag is used to link against the libdl library in Unix-like systems. This library provides an interface for dynamic loading of shared libraries at runtime, which includes functions like dlopen, dlsym, dlclose, and dlerror. These functions allow a program to load shared libraries dynamically, resolve symbols within them, and unload them when they are no longer needed.
I think it will be impossible to get rid of this library. What do you think?/